### PR TITLE
Capture Cabin Overheat protection status

### PR DESCRIPTION
**Pull request by cwanja**
Full disclosure - like others - this is not my area of expertise, but I studied other PRs and hope that I got at least it started and it is not all trash 🤣 

Following discussion #3316, adding the API endpoints / data to capture when cabin overheat protection is on. Once these are added, update `Trip` (new panel) to capture temperatures and when / if cabin overheat protection was kicked in.

### CODE REVIEW

File: lib/tesla_api/vehicle/state.ex [hunk 1]
```diff
@@ -133,6 +135,8 @@ defmodule TeslaApi.Vehicle.State do
         battery_heater: climate["battery_heater"],
         battery_heater_no_power: climate["battery_heater_no_power"],
         climate_keeper_mode: climate["climate_keeper_mode"],
+        cabin_overheat_protection: climate["cabin_overheat_protection"],
+        cabin_overheat_protection_actively_cooling: claimte["cabin_overheat_protection_actively_cooling"],
```
Comment (jfliu1) [inline lib/tesla_api/vehicle/state.ex:139]: typo: "Claimte" should be 'climate'.

Comment (cwanja) [reply lib/tesla_api/vehicle/state.ex]: Fixed! Thank you 😄

File: lib/teslamate/vehicles/vehicle.ex [hunk 2]
```diff
@@ -1261,6 +1263,8 @@ defmodule TeslaMate.Vehicles.Vehicle do
       passenger_temp_setting: vehicle.climate_state.passenger_temp_setting,
       is_rear_defroster_on: vehicle.climate_state.is_rear_defroster_on,
       is_front_defroster_on: vehicle.climate_state.is_front_defroster_on,
+      cabin_overheat_protection: vehicle.climate_state.cabin_overheat_protection,
+      cabin_overheat_protection_actively_cooling: vehicle.climate_state.cabin_overheat_protection_actively_cooling
```
Comment (jfliu1) [inline lib/teslamate/vehicles/vehicle.ex:1267]: missing comma at the end of line.

Comment (cwanja) [reply lib/teslamate/vehicles/vehicle.ex]: Thank you again! 😄

**Comment by JakobLichterfeld** [comment]
please update your branch

**Comment by SaswatPadhi** [comment]
Thanks @cwanja for opening the PR. I was just about to submit a PR for this, but saw yours as open already :)

Would it be possible to also add cabin overheat protection to MQTT as well, please?
AFAICT, in addition to your changes, it should be a small change to `lib/teslamate/mqtt/pubsub/vehicle_subscriber.ex` as well ...

**Comment by cwanja** [comment]
Hey @SaswatPadhi, I have pretty much abandoned this PR. I can close it without merging it or you can build on top of it. I was overly ambitious when I started and when Adrian handed over, I knew I would not be able to complete.

**Pull request closed without merging**